### PR TITLE
Added Test for Reload

### PR DESCRIPTION
**Pull request by sammychinedu2ky**
Fixes https://github.com/G-Research/consuldotnet/issues/264
@marcin-krystianc @naskio

### CODE REVIEW

**Review by marcin-krystianc — COMMENTED** [review]
There are just a few minor comments to be addressed in the test, but overall it looks good.

File: Consul/Interfaces/IAgentEndpoint.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public interface IAgentEndpoint
         Task<QueryResult<LocalServiceHealth>> GetLocalServiceHealthByID(string serviceID, QueryOptions q, CancellationToken ct = default);
         Task<QueryResult<LocalServiceHealth>> GetLocalServiceHealthByID(string serviceID, CancellationToken ct = default);
         Task<QueryResult<Metrics>> GetAgentMetrics(CancellationToken ct = default);
+        Task<WriteResult> Reload(CancellationToken ct = default);
         Task<WriteResult> Leave(string node, CancellationToken ct = default);
-        Task<WriteResult> Reload(string node, CancellationToken ct = default);
```
[the remark this answers is not in the document]
Comment (sammychinedu2ky) [reply Consul/Interfaces/IAgentEndpoint.cs:67]: alright got that. thanks

**Comment by sammychinedu2ky** [comment]
Thanks done @marcin-krystianc

**Review by marcin-krystianc — COMMENTED** [review]
Hi @sammychinedu2ky,

one more thing. Since we rely on the `CONSUL_AGENT_CONFIG_PATH` to be set, we need to set it in the CI workflow so the test is actually run 😄

**Comment by sammychinedu2ky** [comment]
Alright 👌

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -126,6 +126,7 @@ jobs:
       - name: Run tests
         shell: bash
         run: |
+          export CONSUL_AGENT_CONFIG_PATH=$GITHUB_WORKSPACE/Consul.Test/test_config.json
```
Comment (marcin-krystianc) [inline .github/workflows/ci.yml:129]: Instead of using export, you can use `env` context in line 133

Comment (sammychinedu2ky) [reply .github/workflows/ci.yml]: I switched it now 😄

**Comment by sammychinedu2ky** [comment]
Hi @marcin-krystianc , some test cases passed. And some didn't
In some instances it said
```
 Expected: DEBUG
Actual:   TRACE
```
and in some instances it said 
```
The given key was not present in the dictionary.
```

**Comment by sammychinedu2ky** [comment]
@marcin-krystianc . I think I would use the skippable feature here

**Comment by marcin-krystianc** [comment]
> @marcin-krystianc . I think I would use the skippable feature here

Makes sense.

**Comment by sammychinedu2ky** [comment]
done @marcin-krystianc

**Review by marcin-krystianc — APPROVED** [review]
Thank you 👍

**Pull request merged**